src/Symfony/Component/Serializer/Normalizer/ArrayDenormalizer.php

```diff
@@ -52,6 +52,13 @@ public function denormalize($data, $class, $format = null, array $context = [])
         $class = substr($class, 0, -2);
 
         $builtinType = isset($context['key_type']) ? $context['key_type']->getBuiltinType() : null;
+
+        // Fix a collection that contains the only one element
+        // This is special to xml format only
```

Comment:
This fix was in ObjectNormalizer, I think it makes more sense here, and now with the recursive types its harder to fit into ObjectNormalizer.